### PR TITLE
Add geoshape query

### PR DESCRIPTION
**Pull request by floristenhove**
This PR adds support for the [Geoshape Query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-geo-shape-query.html).

This also adds two enums, which requires a minimum of PHP 8.1. This would mean a breaking change.
If this is not appreciated at this point, I'm happy to change this to use a simple `string`.

### CODE REVIEW

File: src/Queries/GeoshapeQuery.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    public const TYPE_POLYGON = 'polygon';
+    public const TYPE_MULTIPOINT = 'multipoint';
+    public const TYPE_MULTILINESTRING = 'multilinestring';
+    public const TYPE_MULTIPOLYGON = 'multipolygon';
+    public const TYPE_GEOMETRYCOLLECTION = 'geometrycollection';
+    public const TYPE_ENVELOPE = 'envelope';
+    public const RELATION_INTERSECTS = 'intersects';
+    public const RELATION_DISJOINT = 'disjoint';
+    public const RELATION_CONTAINS = 'contains';
+    public const RELATION_WITHIN = 'within';
```
Comment (bram-pkg) [inline src/Queries/GeoshapeQuery.php:20]:
```suggestion
    public const RELATION_WITHIN = 'within';

```

File: src/Queries/GeoshapeQuery.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    public function __construct(
+        string $field,
+        string $type,
+        array $coordinates,
+        ?string $relation = self::RELATION_INTERSECTS,
+    ) {
+        $this->field = $field;
+        $this->coordinates = $coordinates;
+
+        Assert::oneOf($type, [
```
Comment (bram-pkg) [inline src/Queries/GeoshapeQuery.php:36]: Sorry for the back and forth, but I don't believe webmozart's assert library is part of our production dependencies. Could we either remove this or solve it without a dev dependency?

Comment (floristenhove) [reply src/Queries/GeoshapeQuery.php]: No worries, it's my bad. I've updated this now.

**Review by bram-pkg — APPROVED** [review]

**Review by bram-pkg — APPROVED** [review]
Thanks for this contribution! Merged.

**Comment by bram-pkg** [comment]
Released as part of 3.2.0

**Pull request merged**